refactor(range): follow current style guides and fix typescript issues

components is not currently following our current style guides and tsc is returning following errors

```
22 export class Range extends Component<RangeProps> {
                                        ~~~~~~~~~~

Range.tsx(22,38): error TS4020: Extends clause of exported class 'Range' has or is using private name 'RangeProps'.


24   static get props() {
                ~~~~~

Range.tsx(24,14): error TS4038: Return type of public static property getter from exported class has or is using name 'PropOptions' from external module "/Users/mario/ngParty/blaze-elements/node_modules/skatejs/src/ts-typings/api" but cannot be named.


6   static get props() {
               ~~~~~

index.demo.tsx(6,14): error TS4038: Return type of public static property getter from exported class has or is using name 'PropOptions' from external module "/Users/mario/ngParty/blaze-elements/node_modules/skatejs/src/ts-typings/api" but cannot be named.
```

--- a/packages/range/Range.test.tsx
+++ b/packages/range/Range.test.tsx
@@ -1,16 +1,63 @@
-import { Range } from './Range';
-
-import { h, mount } from 'bore';
+import { mount, h } from 'bore';
+import * as expect from 'expect';
+import { Range } from './index';
 
 describe( Range.is, () => {
 
-  it( 'should render', () => {
+  describe( `Custom element`, () => {
 
-    console.warn( 'Missing tests for Range!' );
+    it( `should be registered`, () => {
 
-    return mount(
-      <Range value={10} />
-    ).wait();
+      expect( customElements.get( Range.is ) ).toBe( Range );
+
+    } );
+
+    it( `should render via JSX IntrinsicElement`, () => {
+
+      return mount(
+        <bl-range />
+      ).wait(( element ) => {
+
+        expect( element.node.localName ).toBe( Range.is );
+
+      } );
+
+    } );
+
+    it( `should render via JSX class`, () => {
+
+      return mount(
+        <bl-range />
+      ).wait(( element ) => {
+
+        expect( element.has( '.c-range' ) ).toBe( true );
+
+      } );
+
+    } );
+  } );
+
+  describe( `API`, () => {
+
+    describe( `[value]`, () => {
+      // TODO
+    } );
+
+    describe( `[min]`, () => {
+      // TODO
+    } );
+
+    describe( `[max]`, () => {
+      // TODO
+    } );
+
+    describe( `[disabled]`, () => {
+      // TODO
+    } );
+
+    describe( `[color]`, () => {
+      // TODO
+    } );
 
   } );
 

--- a/packages/range/Range.tsx
+++ b/packages/range/Range.tsx
@@ -1,56 +1,85 @@
-import { h, Component, prop, emit } from 'skatejs';
+import { h, Component, emit, props } from 'skatejs';
 import styles from './Range.scss';
-import { ColorType, cssClassForColorType, css } from '@blaze-elements/common';
+import {
+  ColorType,
+  cssClassForColorType,
+  css,
+  shadyCssStyles,
+  prop
+} from '@blaze-elements/common';
 
+export type RangeProps = Props & EventProps;
 
-// public
-type RangeProps = Props & EventProps;
-type Props = {
-  value: number | string,
+export type Attrs = {
+  value?: number | string,
   min?: number | string,
   max?: number | string,
   disabled?: boolean,
   color?: ColorType,
 };
-type EventProps = {
+
+export type Props = {
+  value?: number | string,
+  min?: number | string,
+  max?: number | string,
+  disabled?: boolean,
+  color?: ColorType,
+};
+
+export type EventProps = {
   onKeyup?: typeof HTMLElement.prototype.onkeyup,
   onFocus?: typeof HTMLElement.prototype.onfocus,
   onBlur?: typeof HTMLElement.prototype.onblur,
   onChange?: ( ev: CustomEvent ) => void,
 };
 
-export class Range extends Component<RangeProps> {
-  static get is() { return 'bl-range'; }
-  static get props() {
-    return {
-      value: prop.number( {
-        attribute: true
-      } ),
-      min: prop.number( {
-        attribute: true
-      } ),
-      max: prop.number( {
-        attribute: true
-      } ),
-      color: prop.string( {
-        attribute: true
-      } ),
-      disabled: prop.boolean( {
-        attribute: true
-      } ),
-    };
-  }
+export type Events = {};
+
+@shadyCssStyles()
+export default class Range extends Component<RangeProps> {
+
   static get events() {
     return {
       change: 'change'
     };
   }
 
-  color: ColorType;
-  value: number;
-  min = 0;
-  max = 100;
-  disabled: boolean;
+  get css() { return styles; }
+
+  @prop( {
+    type: String,
+    attribute: {
+      source: true
+    }
+  } ) color: ColorType;
+
+  @prop( {
+    type: Number,
+    attribute: {
+      source: true
+    }
+  } ) value: number;
+
+  @prop( {
+    type: Number,
+    attribute: {
+      source: true
+    }
+  } ) min = 0;
+
+  @prop( {
+    type: Number,
+    attribute: {
+      source: true
+    }
+  } ) max = 100;
+
+  @prop( {
+    type: Boolean,
+    attribute: {
+      source: true
+    }
+  } ) disabled: boolean;
 
   inputElement: HTMLInputElement;
 
@@ -65,8 +94,7 @@ export class Range extends Component<RangeProps> {
     const colorClass = cssClassForColorType( 'c-range', color );
     const className = css( 'c-range', colorClass );
 
-    return [
-      <style>{styles}</style>,
+    return (
       <input
         ref={this.setInputElementRef}
         className={className}
@@ -77,19 +105,14 @@ export class Range extends Component<RangeProps> {
         onChange={this.provideValue}
         disabled={disabled}
       />
-    ];
+    );
   }
 
   private provideValue( event: Event ) {
-    this.value = this.inputElement.valueAsNumber;
+    props( this, { value: this.inputElement.valueAsNumber } );
     emit( this, Range.events.change ); // emit change event on root element
   }
   private setInputElementRef( ref: HTMLInputElement ) {
     this.inputElement = ref;
   }
 }
-
-
-customElements.define( Range.is, Range );
-
-

--- a/packages/range/index.demo.tsx
+++ b/packages/range/index.demo.tsx
@@ -1,14 +1,17 @@
-import { h, Component, prop } from 'skatejs';
-import { Range } from './Range';
+import { h, Component, props } from 'skatejs';
+import { prop, customElement } from '@blaze-elements/common';
+import { Range } from './index';
 
-export class Demo extends Component<void> {
-  static get is() { return 'bl-range-demo'; }
-  static get props() {
-    return {
-      val: prop.number()
-    };
-  }
+export type RangeDemoProps = {
+  val?: number
+};
 
+@customElement( 'bl-range-demo' )
+export class Demo extends Component<RangeDemoProps> {
+
+  @prop( {
+    type: Number,
+  } )
   val: number;
 
   connectedCallback() {
@@ -17,8 +20,7 @@ export class Demo extends Component<void> {
   }
 
   renderCallback() {
-    return [
-      <style />,
+    return (
       <fieldset>
         <legend>{Range.is}</legend>
 
@@ -73,7 +75,7 @@ export class Demo extends Component<void> {
         </div>
 
       </fieldset>
-    ];
+    );
   }
 
   private log = ( message: string ): any => {
@@ -81,8 +83,8 @@ export class Demo extends Component<void> {
   }
 
   private reflectValue( event: Event ) {
-    this.val = Number.parseInt(( event.currentTarget as HTMLInputElement ).value );
+    props( this, {
+      val: Number.parseInt(( event.currentTarget as HTMLInputElement ).value )
+    } );
   }
 }
-
-customElements.define( Demo.is, Demo );

--- a/packages/range/index.ts
+++ b/packages/range/index.ts
@@ -1,1 +1,19 @@
-import './Range';
+
+import { customElement, GenericTypes } from '@blaze-elements/common';
+import RawRange, { RangeProps, Attrs, Events } from './Range';
+
+const Range = customElement( 'bl-range' )( RawRange ) as typeof RawRange;
+
+export {
+  Range
+};
+
+
+declare global {
+  namespace JSX {
+    interface IntrinsicElements {
+      'bl-range': GenericTypes.IntrinsicCustomElement<RangeProps>
+      & GenericTypes.IntrinsicBoreElement<Attrs, Events>
+    }
+  }
+}

